### DISCUSSION TITLE
train_unet_demo produces wrong image

### DISCUSSION THREAD

**Discussion by khlee0192** [facebookresearch/fastMRI#290] [General]
I am working on applying different loss functions on unet module.

To do that, I tried to run train_unet_demo.py file with channel of 128
(the leaderboard code has 256, but I had to decrease channels due to GPU memory)

However, tested images comes out to be very strange. Providing my image and ground truth image, my image seems to have some kind of shape but not the exact type.

I tried to fix this by changing color map or manually calculating pixel values, but did not work well. Could you give any idea to solve this issue?

* GPU : 3060 Ti 8GB RAM
* I don't know exact source of dataset, since it is given by my supervisor.

<img width="345" alt="스크린샷 2023-03-03 오후 12 25 21" src="https://user-images.githubusercontent.com/115933099/222625621-f3450fea-017e-45eb-a01e-e8f2a06905e2.png">
<img width="202" alt="스크린샷 2023-03-03 오후 12 25 37" src="https://user-images.githubusercontent.com/115933099/222625638-6c55261a-c4b1-4a8c-973d-a674beadc2f6.png">

**Comment by khlee0192** [comment]
Though I reported this issue as bug, I think it would be problem on my code.

**Reply by mmuckley** [reply]
Hello @khlee0192, this does look like a normalization issue to me. The aspect ratios also seem to be off. I wonder if you need to transpose the image.

**Reply by khlee0192** [reply]
Thank you, I will give it a try

**Reply by ChunLiu-Claire** [reply]
> Thank you, I will give it a try

Hello, have you solved this problem yet? I am currently experiencing the same problem.

**Reply by khlee0192** [reply]
I didn't have enough time to work on this problem.. and still could not solve this issue. Still having strange image.

**Comment by mmuckley** [comment]
@purerabbit @khlee0192 could one of you try using the [pretrained inference script](https://github.com/facebookresearch/fastMRI/blob/main/fastmri_examples/unet/run_pretrained_unet_inference.py) on your data? In particular, also make sure your preprocessing matches the [UNet DataTransform](https://github.com/facebookresearch/fastMRI/blob/0c9fd544c1dc2e5de0b576ab0fb0349c1225924b/fastmri/data/transforms.py#L282-L389).

**Comment by marinapapageorgiou** [comment]
Hello @mmuckley,
I run the pretrained model (run_pretrained_unet_inference.py) and the results are these images:
![1](https://github.com/user-attachments/assets/45646138-510c-48d4-9013-575ad0469fbb)

![2](https://github.com/user-attachments/assets/95bc0bcb-ee27-417c-8166-7a94654fc233)


After I opened the images of testing (train_unet_demo.py) that the Unet model gave me and the results are these:
![3](https://github.com/user-attachments/assets/b0422dc2-c27d-47c6-b52f-e7b20def871b)

![4](https://github.com/user-attachments/assets/33ccfd82-3487-4af8-8895-2ec40bad10df)


If it is usuful I run the model with max_epochs=50 and lr=0.001,  # learning rate
Also, I am using only these files for dataset 
knee_singlecoil_train
knee_singlecoil_val
knee_singlecoil_test

So my question is, What is wrong with the testing images of my training model?
